### PR TITLE
fixed Boolean error but new trace error appeared!

### LINKED ISSUE #5: fix pararm combine function boolean error

### PR DESCRIPTION
**Pull request by Minijanach**

### CODE REVIEW

**Comment by Minijanach** [comment]
Fixed all the Boolean errors by playing around with jnp.where().
New trace error appears and i m really not sure how to fix it.
Tried fixing the value with partial tried converting it from a traced value to an integer, currently i dont have any clue how to solve it.
Error Message:
Screenshot from 2024-11-07 05-26-57

**Comment by prashjet** [comment]
Hi Marco. I'm not sure how to solve exactly that bug, but here is a suggestion for an alternative approach that might be simpler.

The errors all seem be related to way you're using the `optimize_mask` to fix some parameters. `ScipyBoundedMinimize` can accept bounds, and the [docs](https://docs.scipy.org/doc/scipy-1.12.0/reference/generated/scipy.optimize.Bounds.html
) say that you can set lower bound and upper bounds to the same value to fix a parameter. That's for the `scipy` function but I guess this should also work for the `jax.scipy` equivalent.

**Comment by Minijanach** [comment]
@prashjet thanks for the suggestion, I will try it out. I'm not sure if the current error is related to the optimized mask because it think that should actually work with the changes I implemented.  It seems to have some problems when creating a grid during the optimization. Linspace wants concrete values which makes sense but fixing these values using partial didn't change anything. 

On thing that came to mind would be to pass the grid as an argument to the function we want to optimize.
I will get back to you when i tried the two approaches, hopefully that will solve the problem.
